### PR TITLE
fix: (Platform) fix compact bug in multi input component

### PR DESCRIPTION
**Pull request by Lokanathan-k**
updating the documentation and addressing bug for compact

#### Please provide a link to the associated issue.

fixes: https://github.com/SAP/fundamental-ngx/issues/4745
fixes: https://github.com/SAP/fundamental-ngx/issues/4781
fixes: https://github.com/SAP/fundamental-ngx/issues/4988

#### Please provide a brief summary of this pull request.

added autofocs to the input component to handling default focus on load.
fixed the compact issue.
removed the unwanted documentation in the file.
removing x mark on the clicking of the input group button.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 1]
```diff
@@ -59,11 +53,13 @@ import { MULTIINPUT_COMPONENT } from './multi-input.interface';
     ]
 })
 export class PlatformMultiInputComponent extends BaseMultiInput implements OnInit, AfterViewInit {
-
     /** type Represent the type of input used for the multi Input */
     @Input()
     type: InputType;
 
+    @Input()
+    autofocus: boolean;
```
Comment (manu-kr) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:61]: Add typedocs for the fields

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 2]
```diff
@@ -125,24 +122,26 @@ export class PlatformMultiInputComponent extends BaseMultiInput implements OnIni
...
-        @Optional() @Self() readonly ngControl: NgControl,
-        @Optional() @Self() readonly ngForm: NgForm,
-        @Optional() readonly dialogConfig: DialogConfig,
-        readonly _dynamicComponentService: DynamicComponentService,
-        @Optional() @Inject(DATA_PROVIDERS) private providers: Map<string, DataProvider<any>>,
-        readonly _listConfig: ListConfig,
-        private _rtlService: RtlService,
-        @Optional() @SkipSelf() @Host() formField: FormField,
-        @Optional() @SkipSelf() @Host() formControl: FormFieldControl<any>
+        /** @hidden */ readonly cd: ChangeDetectorRef,
```
Comment (manu-kr) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:134]: Check code formatting, after '/**@hidden*/' there should be a new line.

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.html [hunk 3]
```diff
@@ -67,6 +68,7 @@
                 [attr.aria-expanded]="isOpen && _suggestions.length > 0"
                 [readonly]="readonly"
                 [attr.aria-readonly]="readonly"
+                autofocus="autofocus"
```
Comment (manu-kr) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.html:71]: arai-label and title is missing for the input field

**Review by Betrozov — APPROVED** [review]

**Comment by JKMarkowski** [comment]
I see there are no hover effects and checkbox elements inside multi input - is it on purpose?

Peek 2021-03-01 14-22

**Comment by Lokanathan-k** [comment]
> I see there are no hover effects and checkbox elements inside multi input - is it on purpose?
> 
> Peek 2021-03-01 14-22

@JKMarkowski since core has the multi input with checkbox, as discussed on the tech spec we will concentrating only on the different type of list  option. addressed the hover effect to the list

**Review by Betrozov — APPROVED** [review]

**Review by sKudum — APPROVED** [review]

**Comment by nikvalor** [comment]
Project: allure-results CORE 
 Url: https://allure-results--core-pr-4764-da40iudb.web.app 
 This link will expire at Sun 04 Apr 2021 11:20:48 AM GMT

**Comment by nikvalor** [comment]
Project: allure-results Platform B
 Url: https://allure-results--platform-b-pr-4764-wu36oboo.web.app 
 This link will expire at Sun 04 Apr 2021 11:33:06 AM GMT

**Comment by nikvalor** [comment]
Project: allure-results Platform A 
 Url: https://allure-results--platform-a-pr-4764-11m907un.web.app 
 This link will expire at Sun 04 Apr 2021 12:50:26 PM GMT

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Review by droshev — CHANGES_REQUESTED** [review]
Can you update the title by writing what has been fixed (e.g. `fix: fix the compact bug in action bar`) ?

When you quote issues that you are fixing you need to write something like:
```
fixes: #1234
```

**Review by JKMarkowski — APPROVED** [review]

**Review by KevinOkamoto — APPROVED** [review]

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform A 
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null Platform B
 Url: null 
 This link will expire at

**Comment by nikvalor** [comment]
Project: null CORE 
 Url: null 
 This link will expire at

File: libs/platform/src/lib/components/base.ts [hunk 4]
```diff
@@ -73,11 +73,11 @@ export abstract class BaseComponent implements OnInit, OnDestroy {
         const injector = PlatformConfig.getInjector();
         this._contentDensityService = injector?.get(ContentDensityService, undefined, InjectFlags.Optional);
         this._router = injector?.get(Router);
-        this._router?.events.subscribe(event => {
+        this._router?.events.subscribe((event) => {
```
Comment (droshev) [inline libs/platform/src/lib/components/base.ts:76]: not needed change

File: libs/platform/src/lib/components/base.ts [hunk 5]
```diff
@@ -93,10 +93,12 @@ export abstract class BaseComponent implements OnInit, OnDestroy {
     /** @hidden */
     private _setupDensitySubscriptions(): void {
         if (this._contentDensity === undefined && this._contentDensityService) {
-            this._subscriptions.add(this._contentDensityService._contentDensityListener.subscribe(density => {
-                this.contentDensity = density;
-                this.markForCheck();
-            }));
+            this._subscriptions.add(
```
Comment (droshev) [inline libs/platform/src/lib/components/base.ts:96]: can you revert this?

File: libs/platform/src/lib/components/form/base.input.ts [hunk 6]
```diff
@@ -37,12 +37,18 @@ let randomId = 0;
  * that e.g. placeholder will change after component is created
  */
 @Directive()
-export abstract class BaseInput extends BaseComponent
+export abstract class BaseInput
```
Comment (droshev) [inline libs/platform/src/lib/components/form/base.input.ts:40]: all those changes in this file should be done in a separate PR where the documentation/formatting is addressed

File: libs/platform/src/lib/components/form/collection-base.input.ts [hunk 7]
```diff
@@ -43,15 +43,15 @@ export abstract class CollectionBaseInput extends BaseInput {
      */
     @Input()
     displayKey: string;
-
+    /** @hidden */
```
Comment (droshev) [inline libs/platform/src/lib/components/form/collection-base.input.ts:46]: same as above

File: libs/platform/src/lib/components/form/multi-input/base-multi-input.ts [hunk 8]
```diff
@@ -298,13 +286,21 @@ export abstract class BaseMultiInput extends CollectionBaseInput implements Afte
     };
 
     constructor(
+        /** @hidden */
```
Comment (droshev) [inline libs/platform/src/lib/components/form/multi-input/base-multi-input.ts:289]: same as above

**Review by droshev — CHANGES_REQUESTED** [review]
Looks like this PR is addressing different things. Can you keep the changes addressing the one issue at the time? otherwise it is not easy to track what this PR is about.

**Comment by puru-hk** [comment]
Focus for the input field is not as per the fiori3 spec 
screenshot : 
image

Expected: 
image

File: libs/platform/src/lib/components/base.ts [hunk 9]
```diff
@@ -93,10 +93,11 @@ export abstract class BaseComponent implements OnInit, OnDestroy {
     /** @hidden */
     private _setupDensitySubscriptions(): void {
         if (this._contentDensity === undefined && this._contentDensityService) {
-            this._subscriptions.add(this._contentDensityService._contentDensityListener.subscribe(density => {
-                this.contentDensity = density;
-                this.markForCheck();
-            }));
+            this._subscriptions.add(this._contentDensityService._contentDensityListener.subscribe((density) => {
```
Comment (droshev) [inline libs/platform/src/lib/components/base.ts:96]: I think this is just a code reformatting, isn't it? If so, could you revert it?

File: libs/platform/src/lib/components/form/base.input.ts [hunk 10]
```diff
@@ -39,10 +39,15 @@ let randomId = 0;
 @Directive()
 export abstract class BaseInput extends BaseComponent
     implements FormFieldControl<any>, ControlValueAccessor, OnInit, OnChanges, DoCheck, AfterViewInit, OnDestroy {
+
```
Comment (droshev) [inline libs/platform/src/lib/components/form/base.input.ts:42]: revert the file

File: libs/platform/src/lib/components/form/multi-input/base-multi-input.ts [hunk 11]
```diff
@@ -136,10 +127,6 @@ export abstract class BaseMultiInput extends CollectionBaseInput implements Afte
     @Input()
     secondaryTextAlignment: TextAlignment = 'right';
 
-    /** Turns on/off Adjustable Width feature */
```
Comment (droshev) [inline libs/platform/src/lib/components/form/multi-input/base-multi-input.ts:139]: was this an intentional input remove? If so then it has to be a breaking change PR

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 12]
```diff
@@ -88,7 +83,8 @@ export class PlatformMultiInputComponent extends BaseMultiInput implements OnIni
     get isGroup(): boolean {
         return !!(this.group && this.groupKey);
     }
-    /** Whether the input is disabled. */
+    /** @hidden
```
Comment (droshev) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:86]: can we revert this change ?

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 13]
```diff
@@ -126,23 +119,36 @@ export class PlatformMultiInputComponent extends BaseMultiInput implements OnIni
     @ViewChild(CdkConnectedOverlay)
     _connectedOverlay: CdkConnectedOverlay;
 
+    /** @hidden */
```
Comment (droshev) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:122]: and those below

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 14]
```diff
@@ -126,23 +119,36 @@ export class PlatformMultiInputComponent extends BaseMultiInput implements OnIni
...
         readonly _dynamicComponentService: DynamicComponentService,
+        /** @hidden */
         @Optional() @Inject(DATA_PROVIDERS) private providers: Map<string, DataProvider<any>>,
+        /** @hidden */
         readonly _listConfig: ListConfig,
+        /** @hidden */
         private _rtlService: RtlService,
+        /** @hidden */
         @Optional() @SkipSelf() @Host() formField: FormField,
+        /** @hidden */
```
Comment (droshev) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:151]: ...until here

**Review by droshev — CHANGES_REQUESTED** [review]
few points

**Comment by puru-hk** [comment]
All the below issues is fixed 
#4745
#4781
#4988

**Review by puru-hk — APPROVED** [review]

**Review by droshev — APPROVED** [review]
next time create a PR for each issue

**Pull request merged**